core/dbt/adapters/base/impl.py

```diff
@@ -1185,3 +1194,24 @@ def catch_as_completed(
             # exc is not None, derives from Exception, and isn't ctrl+c
             exceptions.append(exc)
     return merge_tables(tables), exceptions
+
+
+def log_code_execution(code_execution_function):
```

Comment:
I tested this decorator in dbt-spark and it works fine, any suggestions on adding a test for it? Maybe just a unittest with a lot of mocking?